File: 151. Reverse Words in a String/151.go
// 151. Reverse Words in a String solution in GoLang
// https://leetcode.com/problems/reverse-words-in-a-string/description/

package leetcode

func reverseWords(s string) string {
    var result string
    var word string

    for i := 0; i < len(s); i++ {
        if s[i] == ' ' {
            if len(word) > 0 && len(result) > 0 {
                result = word + " " + result;
            } else if len(word) > 0 {
                result = word;
            }
            word = ""
        } else {
            word = word + string(s[i])
        }
    }
    
    if len(word) > 0 && len(result) > 0 {
                result = word + " " + result;
            } else if len(word) > 0 {
                result = word;
            }

    return result
}

File: 151. Reverse Words in a String/151.js
// 151. Reverse Words in a String solution in JavaScript
// https://leetcode.com/problems/reverse-words-in-a-string/description/

/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let result = [];
    let word = [];

    for (let i = 0; i < s.length; i++) {
        if (s.charAt(i) == ' ') {
            if (word.length > 0) result.unshift(word.join(''));
            word = [];
        } else {
            word.push(s.charAt(i));
        }
    }

    if (word.length > 0)  result.unshift(word.join(''));

    return result.join(' ');
};